### PR TITLE
fix: panic on e2e tests

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

# Proposed changes

Fixes a panic occuring during e2e tests

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

### CODE REVIEW

File: pkg/repository/git.go [hunk 1]
```diff
@@ -32,27 +34,34 @@ func newGitRepoController() gitRepoController {
...
-func (r gitRepoController) isClean() (bool, error) {
-	repo, err := r.repoGetter.get(r.path)
-	if err != nil {
-		return false, fmt.Errorf("failed to analyze git repo: %w", err)
-	}
-	worktree, err := repo.Worktree()
-	if err != nil {
-		return false, fmt.Errorf("failed to infer the git repo's current branch: %w", err)
+func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Second)
```
Comment (jLopezbarb) [inline pkg/repository/git.go:38]: Fixes #2970

File: integration/deploy/compose_test.go [hunk 2]
```diff
@@ -235,7 +236,7 @@ func TestReDeployPipelineFromCompose(t *testing.T) {
 		Token:      token,
 	}
 	require.NoError(t, commands.RunOktetoCreateNamespace(oktetoPath, namespaceOpts))
-	defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
+	// defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
```
Comment (AdrianPedriza) [inline integration/deploy/compose_test.go:239]: we should remove the namespace, right?

File: integration/deploy/compose_test.go [hunk 3]
```diff
@@ -156,7 +156,7 @@ func TestDeployPipelineFromCompose(t *testing.T) {
...
```
Comment (AdrianPedriza) [inline integration/deploy/compose_test.go:159]: we should remove the namespace, right?

File: integration/deploy/compose_test.go [hunk 4]
```diff
@@ -315,7 +317,7 @@ func TestDeployPipelineFromComposeOnlyOneSvc(t *testing.T) {
...
```
Comment (AdrianPedriza) [inline integration/deploy/compose_test.go:320]: we should remove the namespace, right?

File: integration/deploy/compose_test.go [hunk 5]
```diff
@@ -375,7 +378,7 @@ func TestDeployPipelineFromOktetoStacks(t *testing.T) {
...
```
Comment (AdrianPedriza) [inline integration/deploy/compose_test.go:381]: we should remove the namespace, right?

File: integration/deploy/compose_test.go [hunk 6]
```diff
@@ -439,7 +443,7 @@ func TestDeployComposeFromOktetoManifest(t *testing.T) {
...
```
Comment (AdrianPedriza) [inline integration/deploy/compose_test.go:446]: we should remove the namespace, right?

File: integration/deploy/endpoints_test.go [hunk 7]
```diff
@@ -156,7 +156,7 @@ func Test_EndpointsFromOktetoManifest_InferredName(t *testing.T) {
 		Token:      token,
 	}
 	require.NoError(t, commands.RunOktetoCreateNamespace(oktetoPath, namespaceOpts))
-	defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
+	// defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
```
Comment (AdrianPedriza) [inline integration/deploy/endpoints_test.go:159]: we should remove the namespace, right?

File: integration/deploy/endpoints_test.go [hunk 8]
```diff
@@ -218,14 +219,15 @@ func Test_EndpointsFromOktetoManifest_Name(t *testing.T) {
...
```
Comment (AdrianPedriza) [inline integration/deploy/endpoints_test.go:222]: we should remove the namespace, right?

File: pkg/repository/git.go (hunk 1 above)
Comment (AdrianPedriza) [reply pkg/repository/git.go:38]: we are avoiding getting stuck but we should investigate the reason to face this timeout I think

**Review by AdrianPedriza — APPROVED** [review]

**Pull request merged**